apps/files_sharing/lib/Controller/ShareAPIController.php

```diff
@@ -181,7 +181,7 @@ protected function formatShare(\OCP\Share\IShare $share, Node $recipientNode = n
 					throw new NotFoundException();
 				}
 			} else {
-				$node = $nodes[0];
+				$node = reset($nodes);
```

Comment:
https://github.com/nextcloud/server/blob/023d028c571720df648a3bf809cefc774d2aba16/apps/comments/lib/Activity/Listener.php#L97-L102

We use array_shift most of the times. maybe you use that as well, just to stay in line

Reply:
For me it's ok, `array_shift` will do the job.

Reply:
Would you like to leave it as `reset`?

Reply:
`reset` is faster than `array_shift` ;)

Reply:
Well then all occurances should be adjusted...

Reply:
It depends ;) If you want to reuse the array pick array_shift because it will update numeric keys. For return the first element reset is "better". Anyway even if we apply all possible micro optimizations I don't think there is a measurable improvement. I guess it was not clear to @waleczny that you asked him to use array_shift instead of reset ;)

```suggestion
				$node = array_shift($nodes);
```